jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -1233,11 +1233,22 @@ public ForcedCharacterEncodingHttpContent(HttpContent content, String characterE
         {
             super(content);
             this.characterEncoding = characterEncoding;
-            String mimeType = content.getContentTypeValue();
-            int idx = mimeType.indexOf(";charset");
-            if (idx >= 0)
-                mimeType = mimeType.substring(0, idx);
-            this.contentType = mimeType + ";charset=" + this.characterEncoding;
+            if (content.getResource().isDirectory())
```

Comment:
At this point, yes, because there is an explicit null check in the method that calls the `ForcedCharacterEncodingHttpContent` constructor to take a different branch.
But I'm going to add a specific null check here for clarity.